### PR TITLE
Pin the pytest-redis requirement for unit tests

### PR DESCRIPTION
**Pull request by lubosmj**
[noissue]

### CODE REVIEW

**Review by dkliban — APPROVED** [review]

**Comment by lubosmj** [comment]
This should resolve recent errors from the lowerbounds runner:
```
E           mirakuru.exceptions.ProcessExitedWithError: The process invoked by the <pytest_redis.executor.process.RedisExecutor: "/usr/bin/redis-server --daemonize no --rdbcompression no --rdbchecksum no --appendonly no --databases 8 --timeout 30 --pidfile /tmp/pytest-of-pulp/pytest-0/pytest-redis-redis_proc0/redis-server.31926.pid --unixsocket /tmp/pytest-of-pulp/pytest-0/pytest-redis-redis_proc0/redis.31926.sock --dbfilename dump.31926.rdb --logfile /tmp/pytest-of-pulp/pytest-0/pytest-redis-redis_proc0/redis-server.31926.log --loglevel notice --syslog-enabled no --bind 127.0.0.1 --port 31926 --dir /tmp/pytest-of-pulp/pytest-0/pytest-redis-redis_proc0 --loadmodule ''" 0x7f3fdfd0fdf0> executor has exited with a non-zero code: 1.
```

**Comment by bmbouter** [comment]
Are we pinning temporarily? Should we use <= or >= instead of ==? Pinning often loads to plugn-to-plugin incompatibility issues.

**Comment by dkliban** [comment]
THis is likely temporary. This is a unit test requirement.

**Comment by bmbouter** [comment]
Are we incompatible with newer versions? What's the issue that we're pinning around? Typically if we pin I would expect a:

`pytest-redis==3.0.2  # http://thelinktotheissue/`

**Comment by dkliban** [comment]
@lubosmj let's make it <3.1.0

**Comment by dkliban** [comment]
> Are we incompatible with newer versions? What's the issue that we're pinning around? Typically if we pin I would expect a:
> 
> `pytest-redis==3.0.2 # http://thelinktotheissue/`

We haven't filed a bug yet. `pytest-redis` 3.1.0 came out yesterday and it already has 2 issues about backwards incompatibility filed. None of them are obviously the same as what we are experiencing.

File: unittest_requirements.txt [hunk 1]
```diff
@@ -4,4 +4,4 @@ aiotools
 pytest<8
 pytest-django
 pytest-asyncio
-pytest-redis
+pytest-redis==3.0.2
```
Comment (bmbouter) [inline unittest_requirements.txt:7]:
```suggestion
pytest-redis<3.1.0  # https://github.com/ClearcodeHQ/pytest-redis/issues/679
```

**Review by dkliban — APPROVED** [review]

**Review by bmbouter — APPROVED** [review]
Thank you!

**Pull request merged**